crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -45,7 +52,39 @@ impl_downcast!(Stage);
 ///
 /// The checker may report a system more times than the amount of constraints it would actually need
 /// to have unambiguous order with regards to a group of already-constrained systems.
-pub struct ReportExecutionOrderAmbiguities;
+pub struct ReportExecutionOrderAmbiguities {
+    level: AmbiguityReportLevel,
```

Comment:
I would be fine just exposing this field directly. That would save on boilerplate and the enum will give us a type-safe API.

Is there some reason why this isn't safe?

Reply:
On the last commit, I made it public, so users which wanna more control over `ReportExecutionOrderAmbiguities` may create the resource directly. I've also kept the `.off()`, `.minimal()` and `.verbose()` to keep things easier to user (at least I think so)

```rust
pub struct ReportExecutionOrderAmbiguities {
    pub level: AmbiguityReportLevel,
    pub ignore_crates: Vec<String>,
}
```

Reply:
> I've also kept the .off(), .minimal() and .verbose() to keep things easier to user (at least I think so)

I'd like to avoid API duplication where possible. I'd definitely accept it if Cart feels otherwise in a final review though, so don't worry about it for now.